### PR TITLE
List projects with filter and sort_by

### PR DESCRIPTION
**Pull request by honnix**
# TL;DR
List projects with filter and sort_by.

## Type
 - [x] Bug Fix
 - [x] Feature
 - [ ] Plugin

## Are all requirements met?

 - [x] Code completed
 - [x] Smoke tested
 - [x] Unit tests added
 - [x] Code documentation added
 - [ ] Any pending items have an associated Issue

## Complete description
`flyte-cli` currently doesn't support listing projects. This PR adds support of that.

This PR also fixes the mismatching non-equal filter between flytekit and flyteadmin.

## Tracking Issue
https://github.com/lyft/flyte/issues/625
https://github.com/lyft/flyte/issues/418

## Follow-up issue
_NA_

### CODE REVIEW

**Comment by honnix** [comment]
@katrogan I think you have more context around this change. Could you please take a look? Thanks. One thing I'm not sure is what information we should print when listing projects.

**Review by wild-endeavor — DISMISSED** [review]

**Comment by wild-endeavor** [comment]
@anandswaminathan can you approve this and merge it please?

**Review by wild-endeavor — APPROVED** [review]

**Comment by wild-endeavor** [comment]
@honnix i don't think my approval counts for anything anymore, trying to get someone at lyft to approve.  cc @anandswaminathan

**Comment by honnix** [comment]
Sure. No worries. This is not urgent. Thanks for the approval @wild-endeavor !

**Comment by anandswaminathan** [comment]
+1

**Review by anandswaminathan — APPROVED** [review]

**Comment by honnix** [comment]
@anandswaminathan Thank you!
